src/eredis_client.erl

```diff
@@ -378,6 +387,32 @@ connect(#state{host = Host,
             {ok, schedule_reconnect(unknown, State#state{socket = Socket})};
         Error ->
             Error
+    end;
+
+connect(#state{sentinel = SentinelOptions} = State) ->
+    case whereis(eredis_sentinel) of
+        undefined -> eredis_sentinel:start_link(SentinelOptions);
+        _ -> ok
+    end,
+
+    case eredis_sentinel:get_master() of
+        {ok, Host, Port} ->
+            #state{socket_options = SocketOptions,
+                tls_options = TlsOptions,
+                connect_timeout = ConnectTimeout,
+                auth_cmd = AuthCmd,
+                database = Db } = State,
+            case connect(Host, Port, SocketOptions, TlsOptions,
+                ConnectTimeout, AuthCmd, Db) of
+                {ok, Socket} ->
+                    %% In case the connection terminates immediately (this happens with
+                    %% an expired certificate with TLS 1.3) schedule a reconnect already
+                    %% so that we don't try to reconnect if an error is received before
+                    %% reconnect_sleep milliseconds has elapsed.
+                    {ok, schedule_reconnect(unknown, State#state{socket = Socket, host = Host, port = Port})};
```

Comment:
No need to set host and port in the state. On reconnect, connect(State) will be called, where sentinel is queried for host and port again.

Btw, there is some code duplication for the connect/1 function clauses with and without sentinel. Is it possible to get rid of it? For example something like this pseudo code:

```Erlang
connect(#state{...., host = Host0, port = Port0, sentinel = Sentinel, ....}) ->
    {Host, Port} = case Sentinel of
                       undefined -> {Host0, Port0};
                       SentinelPid -> eredis_sentinel:get_master(SentinelPid)
                   end,
    case connect(Host, Port, ......) of
        {ok, Socket} -> ....
        ConnError -> ConnError
    end.
```